tdrs-backend/tdpservice/parsers/aggregates.py

```diff
@@ -39,22 +40,25 @@ def case_aggregates_by_month(df, dfs_status):
             if isinstance(schema_model, SchemaManager):
                 schema_model = schema_model.schemas[0]
 
-            curr_case_numbers = set(schema_model.document.Django.model.objects.filter(datafile=df)
-                                    .filter(RPT_MONTH_YEAR=rpt_month_year)
+            curr_case_numbers = set(schema_model.document.Django.model.objects.filter(datafile=df,
+                                                                                      RPT_MONTH_YEAR=rpt_month_year)
                                     .distinct("CASE_NUMBER").values_list("CASE_NUMBER", flat=True))
             case_numbers = case_numbers.union(curr_case_numbers)
 
         total += len(case_numbers)
-        cases_with_errors += ParserError.objects.filter(file=df).filter(
-            case_number__in=case_numbers).distinct('case_number').count()
+        cases_with_errors += ParserError.objects.filter(file=df, case_number__in=case_numbers)\
+            .distinct('case_number').count()
         accepted = total - cases_with_errors
 
         aggregate_data['months'].append({"month": month,
                                          "accepted_without_errors": accepted,
                                          "accepted_with_errors": cases_with_errors})
 
-    aggregate_data['rejected'] = ParserError.objects.filter(file=df).filter(case_number=None).distinct("row_number")\
-        .exclude(row_number=0).count()
+    error_type_query = Query(error_type=ParserErrorCategoryChoices.PRE_CHECK) | \
+        Query(error_type=ParserErrorCategoryChoices.CASE_CONSISTENCY)
```

Comment:
@elipe17 does this assign an overall status of "rejected" if the file has any cat 4 errors, even if some records in the file are not associated with cat 4 type errors? or just reject the records that include cat 4 errors?

Reply:
@ADPennington, it does not. That query is strictly counting the number of cases that were rejected throughout the entirety of the file. Since records with Cat1 (PRE_CHECK) errors do not get serialized and entire cases with Cat4 (CASE_CONSISTENCY) errors don't get serialized, they contribute to the overall count of rejected cases. The code below (which can be found [here](https://github.com/raft-tech/TANF-app/blob/develop/tdrs-backend/tdpservice/parsers/models.py#L96)) is what determines the status of the DF.

```python
def get_status(self):
        """Set and return the status field based on errors and models associated with datafile."""
        errors = ParserError.objects.filter(file=self.datafile)

        # excluding row-level pre-checks and trailer pre-checks.
        precheck_errors = errors.filter(error_type=ParserErrorCategoryChoices.PRE_CHECK)\
                                .exclude(field_name="Record_Type")\
                                .exclude(error_message__icontains="trailer")\
                                .exclude(error_message__icontains="Unknown Record_Type was found.")

        case_consistency_errors = errors.filter(error_type=ParserErrorCategoryChoices.CASE_CONSISTENCY)

        row_precheck_errors = errors.filter(error_type=ParserErrorCategoryChoices.PRE_CHECK)\
                                    .filter(field_name="Record_Type")\
                                    .exclude(error_message__icontains="trailer")

        if errors is None:
            return DataFileSummary.Status.PENDING
        elif precheck_errors.count() > 0 or case_consistency_errors.count() > 0 or \
                self.total_number_of_records_created == 0:
            return DataFileSummary.Status.REJECTED
        elif errors.count() == 0:
            return DataFileSummary.Status.ACCEPTED
        elif row_precheck_errors.count() > 0:
            return DataFileSummary.Status.PARTIALLY_ACCEPTED
        else:
            return DataFileSummary.Status.ACCEPTED_WITH_ERRORS
```